src/m365/teams/commands/report/report-pstncalls.ts

```diff
@@ -0,0 +1,93 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+import { TeamsPstnCall } from './TeamsPstnCall';
+
+interface CommandArgs {
+  options: DateTimeOptions;
+}
+
+interface DateTimeOptions extends GlobalOptions {
+  fromDateTime: string;
+  toDateTime?: string;
+}
+class TeamsReportPstnCallsCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_REPORT_PSTNCALLS}`;
+  }
+
+  public get description(): string {
+    return 'Get details about PSTN calls made within a given time period';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'calleeNumber', 'callerNumber', 'startDateTime'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const toDateTimeParameter: string = encodeURIComponent(args.options.toDateTime ? args.options.toDateTime : new Date().toISOString());
+    const endpoint: string = `${this.resource}/beta/communications/callRecords/getPstnCalls(fromDateTime=${encodeURIComponent(args.options.fromDateTime)},toDateTime=${toDateTimeParameter})`;
+    this.executeReport(endpoint, logger, args.options.output, cb);
+  }
+
+  protected executeReport(endPoint: string, logger: Logger, output: string | undefined, cb: () => void): void {
+    const requestOptions: any = {
+      url: endPoint,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .get<{ value: TeamsPstnCall[] }>(requestOptions)
```

Comment:
Since we're not interacting with any of the properties in the response, we can skip the type and define the response as `{ value: any[]; }` for simplicity.